$NetBSD$

1) Use the path for pkgsrc cyclone-bootstrap for the system cyclone.
2) Use pkgsrc libtommath (internally named libcyclonebn).

--- Makefile.orig	2024-02-14 02:31:23.000000000 +0000
+++ Makefile
@@ -9,15 +9,15 @@ include Makefile.config
 # Set up Cyclone here to build the compiler itself using a system-installed
 # compiler (EG: from bootstrap or an earlier cyclone version). Everything
 # else can then be built using our local binary.
-CYCLONE_SYSTEM = cyclone -I . -CLNK '-L.'
-CYCLONE_LOCAL = ./cyclone -I . -I libs -COPT '-Iinclude' -CLNK '-L.'
+CYCLONE_SYSTEM = cyclone-bootstrap -CLNK '-L. -L$(PREFIX)/lib'
+CYCLONE_LOCAL = ./cyclone -I . -I libs -COPT '-Iinclude' -CLNK '-L. -L$(PREFIX)/lib'
 CCOMP = $(CC) $(CFLAGS)
 FORMAT_CMD = indent -linux -l80 -i2 -nut
 
 # Libraries
 CYC_RT_LIB = libcyclone.a
-CYC_BN_LIB = libcyclonebn.a
-CYC_BN_LIB_SUBDIR = third-party/libtommath-1.2.0
+#CYC_BN_LIB = libcyclonebn.a
+#CYC_BN_LIB_SUBDIR = third-party/libtommath-1.2.0
 CYC_LIBS = $(CYC_RT_LIB) $(CYC_BN_LIB)
 
 # Directories
@@ -70,7 +70,7 @@ clean :
 	rm -f tests/srfi-143-tests
 	rm -f tests/macro-hygiene
 	rm -f tests/match-tests
-	cd $(CYC_BN_LIB_SUBDIR) ; $(MAKE) clean
+#	cd $(CYC_BN_LIB_SUBDIR) ; $(MAKE) clean
 
 install : libs install-libs install-includes install-bin
 	$(MKDIR) $(DESTDIR)$(DATADIR)
@@ -188,8 +188,8 @@ icyc : icyc.scm $(CYC_RT_LIB) $(CYC_BN_L
 
 $(CYC_RT_LIB) : $(CFILES) $(HEADERS) $(CYC_BN_LIB)
 
-$(CYC_BN_LIB) : $(CYC_BN_LIB_SUBDIR)/*.c
-	cd $(CYC_BN_LIB_SUBDIR) ; $(MAKE) LIBNAME=$(CYC_BN_LIB) && cp $(CYC_BN_LIB) ../..
+#$(CYC_BN_LIB) : $(CYC_BN_LIB_SUBDIR)/*.c
+#	cd $(CYC_BN_LIB_SUBDIR) ; $(MAKE) LIBNAME=$(CYC_BN_LIB) && cp $(CYC_BN_LIB) ../..
 
 hashset.o : hashset.c $(HEADERS)
 	$(CCOMP) -c $< -o $@
@@ -336,7 +336,7 @@ install-includes : $(HEADER_DIR)/*.h
 install-libs : $(CYC_LIBS)
 	$(MKDIR) $(DESTDIR)$(LIBDIR)
 	$(INSTALL) -m0644 $(CYC_RT_LIB) $(DESTDIR)$(LIBDIR)/
-	$(INSTALL) -m0644 $(CYC_BN_LIB) $(DESTDIR)$(LIBDIR)/
+#	$(INSTALL) -m0644 $(CYC_BN_LIB) $(DESTDIR)$(LIBDIR)/
 
 install-bin : cyclone icyc
 	$(MKDIR) $(DESTDIR)$(BINDIR)
